### PR TITLE
must-gather: noobaa-db pod collection on postgres label

### PR DESCRIPTION
**Pull request by rajatsing**
This PR will collect noobaa-db pod logs based on the Postgres label on it, this change comes after the new addition of postgres label on the noobaa db pod. It will collect the logs in the Noobaa folder as well.
BZ Link: https://bugzilla.redhat.com/show_bug.cgi?id=1915644

Signed-off-by: RAJAT SINGH <rajasing@redhat.com>

### CODE REVIEW

File: deploy/deploy-with-olm.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ spec:
...
     mediatype: image/svg+xml
-  image: quay.io/ocs-dev/ocs-registry:latest
+  image: quay.io/rajasing/ocs-registry:latest
```
Comment (crombus) [inline deploy/deploy-with-olm.yaml:29]: This change doesn’t belong here.

File: must-gather/collection-scripts/gather_noobaa_resources [hunk 2]
```diff
@@ -57,11 +57,18 @@ do
     done
 done
 
+# Collecting noobaa db pod logs with postgres label
+NOOBAA_POSTGRES_LABEL='noobaa-db in (postgres)'
+for pod in $(oc -n get pods -l "${NOOBAA_POSTGRES_LABEL}" | grep -v NAME | aws "{print $1}"); do
```
Comment (crombus) [inline must-gather/collection-scripts/gather_noobaa_resources:62]:
```suggestion
for pod in $(oc -n get pods -l "${NOOBAA_POSTGRES_LABEL}" | grep -v NAME | awk '{print $1}'); do
```

**Review by crombus — CHANGES_REQUESTED** [review]
In commit description divide the line into two.

File: deploy/deploy-with-olm.yaml (hunk 1 above)
Comment (rajatsing) [reply deploy/deploy-with-olm.yaml:29]: Thanks for catching this!.

File: must-gather/collection-scripts/gather_noobaa_resources (hunk 2 above)
Comment (rajatsing) [reply must-gather/collection-scripts/gather_noobaa_resources:62]: It should be double quotes, the operator CI failed saying that, this particular line should have double quotes and not a single quote.

Comment (crombus) [reply must-gather/collection-scripts/gather_noobaa_resources:62]: well we are using single quotes in must-gather scripts so can you try once with single quotes and then run `make shellcheck-test` to verify.

Comment (aruniiird) [reply must-gather/collection-scripts/gather_noobaa_resources:62]: Hi @rajatsing , I think crombus is pointing to your command itself (which should be `awk`). In the same script itself, there are other `awk` commands which have single quotes arguments...

Comment (rajatsing) [reply must-gather/collection-scripts/gather_noobaa_resources:62]: Oh yes!, you are right!. Thanks for pointing this out

File: deploy/deploy-with-olm.yaml (hunk 1 above)
Comment (rajatsing) [reply deploy/deploy-with-olm.yaml]: Updated!.

File: must-gather/collection-scripts/gather_noobaa_resources (hunk 2 above)
Comment (rajatsing) [reply must-gather/collection-scripts/gather_noobaa_resources]: Updated, Thanks @aruniiird @crombus

**Review by jarrpa — CHANGES_REQUESTED** [review]
It looks like there are two unrelated whitespace changes in the commit. Please move them out to a separate commit.

**Comment by rajatsing** [comment]
I have removed the extra line that I added my mistake!.
Please review @jarrpa

File: must-gather/collection-scripts/gather_noobaa_resources [hunk 3]
```diff
@@ -1,5 +1,4 @@
 #!/bin/bash
-
```
Comment (crombus) [inline must-gather/collection-scripts/gather_noobaa_resources:2]: add this line back

File: must-gather/collection-scripts/gather_noobaa_resources [hunk 4]
```diff
@@ -57,11 +56,18 @@ do
     done
 done
 
+# Collecting noobaa db pod logs with postgres label
+NOOBAA_POSTGRES_LABEL='noobaa-db in (postgres)'
+for pod in $(oc -n get pods -l "${NOOBAA_POSTGRES_LABEL}" | grep -v NAME | awk '{print $1}'); do
```
Comment (crombus) [inline must-gather/collection-scripts/gather_noobaa_resources:61]: add ns after -n

**Review by crombus — CHANGES_REQUESTED** [review]

File: must-gather/collection-scripts/gather_noobaa_resources (hunk 4 above)
Comment (rajatsing) [reply must-gather/collection-scripts/gather_noobaa_resources:61]: right!

File: must-gather/collection-scripts/gather_noobaa_resources [hunk 5]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-  
+
```
Comment (rajatsing) [inline must-gather/collection-scripts/gather_noobaa_resources:2]: looks good?.

Comment (crombus) [reply must-gather/collection-scripts/gather_noobaa_resources:2]: yup

**Comment by crombus** [comment]
ok just saw @jarrpa comment 

> It looks like there are two unrelated whitespace changes in the commit. Please move them out to a separate commit.

the spaces you see remove that from this commit add another commit with those changes.

**Comment by jarrpa** [comment]
> > It looks like there are two unrelated whitespace changes in the commit. Please move them out to a separate commit.
> 
> the spaces you see remove that from this commit add another commit with those changes.

The second commit that fixes the erroneous removal of the newline should be merged into the first commit. This still does not address the unrelated change of trimming off the dangling whitespace at the end of line 64/70, which should either be removed or moved to a separate commit.

**Comment by rajatsing** [comment]
@crombus can you fix this and force push?.

**Comment by crombus** [comment]
> @crombus can you fix this and force push?.

done. @jarrpa can you review this.

**Comment by rajatsing** [comment]
pinging @jarrpa  in case you missed! :)

**Review by jarrpa — APPROVED** [review]
Changes look good, and commits look clean!

BTW @rajatsing if you really want to get my attention faster, Chat is the preferred method. :) I already get email notifications for every comment to a PR, so adding one more is not too reliable. :joy: 

/lgtm

**Pull request merged**